libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -180,6 +180,10 @@ AP_Proximity::AP_Proximity(AP_SerialManager &_serial_manager) :
     serial_manager(_serial_manager)
 {
     AP_Param::setup_object_defaults(this, var_info);
+    if (_singleton != nullptr) {
```

Comment:
isn't it better for the compiler to catch this by adding code like [this](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_ADSB/AP_ADSB.h#L40)?

Reply:
gotcha

Reply:
Added the removal of copy to proximity.